markdown/main.go

```diff
@@ -80,30 +80,22 @@ func (mr *MarkdownFmtRenderer) blockCode(node ast.Node, source []byte) {
 
 	lang := ""
 	tnode, isFenced := node.(*ast.FencedCodeBlock)
-	if isFenced {
-		info := tnode.Info
-		if info != nil {
-			lang = string(info.Text(source))
-		}
-	}
-	// Parse out the language name.
-	count := 0
-	for _, elt := range strings.Fields(lang) {
-		if elt[0] == '.' {
-			elt = elt[1:]
-		}
-		if len(elt) == 0 {
-			continue
+	mr.buf.WriteString("```")
+	if isFenced && tnode.Info != nil {
+		mr.buf.Write(tnode.Info.Text(source))
+
+		for _, elt := range strings.Fields(lang) {
+			if elt[0] == '.' {
+				elt = elt[1:]
+			}
```

Comment:
This can be cleaned in one shot using something like:
`elt = strings.TrimSpace(strings.TrimLeft(elt, ". "))`

That could collapse the space after the first word in the language tag, though.  I don't have a quick test case for it.

Reply:
:+1: will do, thanks taking a look!